app/uk/gov/hmrc/agentepayeregistration/controllers/AgentEpayeRegistrationController.scala

```diff
@@ -44,16 +44,17 @@ class AgentEpayeRegistrationController @Inject()(@Named("extract.auth.stride.enr
     }.recoverTotal(_ => Future.successful(BadRequest))
   }
 
-  //TODO APB-1125
-  def extract = Action.async { implicit request =>
+  def extract(dateFrom: String, dateTo: String) = Action.async { implicit request =>
     authorised(Enrolment(strideEnrolment) and AuthProviders(PrivilegedApplication)) {
-        Future.successful(Ok(Json.arr()))
-      }.recoverWith {
-        case ex: NoActiveSession =>
-          Future.successful(Unauthorized(ex.getMessage))
-        case ex: AuthorisationException =>
-          Future.successful(Forbidden(ex.getMessage))
+      service.extract(dateFrom, dateTo).map {
+        case Right(registrations) => Ok(Json.obj("registrations" -> Json.toJson(registrations)))
+        case Left(failure) => BadRequest(Json.toJson(failure))
       }
-
+    }.recoverWith {
+      case ex: NoActiveSession =>
+        Future.successful(Unauthorized(ex.getMessage))
```

Comment:
Not sure we need to expose the error message here, should we only need to log the message?